app/src/main/java/com/eslamwaheed/mvvmdemo/repository/UnsplashRepository.kt

```diff
@@ -1,15 +1,30 @@
 package com.eslamwaheed.mvvmdemo.repository
 
 import com.eslamwaheed.mvvmdemo.api.UnsplashService
-import com.eslamwaheed.mvvmdemo.data.UnsplashSearchResponse
-import kotlinx.coroutines.flow.Flow
+import com.eslamwaheed.mvvmdemo.data.daos.PhotosDAO
+import com.eslamwaheed.mvvmdemo.data.models.UnsplashSearchResponse
 import javax.inject.Inject
 
-class UnsplashRepository @Inject constructor(private val service: UnsplashService) {
+class UnsplashRepository @Inject constructor(
+    private val service: UnsplashService,
+    private val dao: PhotosDAO
+) {
     suspend fun getSearchResultStream(query: String): UnsplashSearchResponse {
```

Comment:
caching is not work until now you had prepared all thing but not use caching when get data from server